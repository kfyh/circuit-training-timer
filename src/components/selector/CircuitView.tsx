import React, { ReactElement } from 'react';
import { Circuit } from '../../types/circuits';

type CircuitProps = {
	circuit: Circuit;
	onClick: (circuit: Circuit) => void;
};

export class CircuitView extends React.Component<CircuitProps, Record<string, never>> {
	constructor(props: CircuitProps) {
		super(props);
	}

	public render(): ReactElement {
		return (
			<div>
				{this.props.circuit.name}
				<button
					onClick={(e) => {
						e.preventDefault();
						this.props.onClick(this.props.circuit);
					}}
				>
					Select
				</button>
			</div>
		);
	}
}