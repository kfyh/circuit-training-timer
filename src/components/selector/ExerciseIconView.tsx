import React, { ReactElement } from 'react';
import { Exercise } from '../../types/circuits';

type ExerciseIconViewProps = {
	exercise: Exercise;
	onSelect: (exercise: Exercise) => void;
	onEdit: (exercise: Exercise) => void;
};

export class ExerciseIconView extends React.Component<ExerciseIconViewProps> {
	constructor(props: ExerciseIconViewProps) {
		super(props);
	}

	public render(): ReactElement {
		return (
			<div>
				{this.props.exercise.name}
				<button
					onClick={(e) => {
						e.preventDefault();
						this.props.onSelect(this.props.exercise);
					}}
				>
					Info
				</button>
				<button
					onClick={(e) => {
						e.preventDefault();
						this.props.onEdit(this.props.exercise);
					}}
				>
					Edit
				</button>
			</div>
		);
	}
}
