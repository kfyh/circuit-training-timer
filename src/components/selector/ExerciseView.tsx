import React, { ReactElement } from 'react';
import { Exercise } from '../../types/circuits';

type ExerciseViewProps = {
	exercise: Exercise;
	onClick: (exercise: Exercise) => void;
};

export class ExerciseView extends React.Component<ExerciseViewProps> {
	constructor(props: ExerciseViewProps) {
		super(props);
	}

	public render(): ReactElement {
		return (
			<div>
				{this.props.exercise.name}
				<button
					onClick={(e) => {
						e.preventDefault();
						this.props.onClick(this.props.exercise);
					}}
				>
					Select
				</button>
			</div>
		);
	}
}
