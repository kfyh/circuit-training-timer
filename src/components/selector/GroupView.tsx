import React, { ReactElement } from 'react';
import { ExerciseGroup } from '../../types/circuits';

type GroupViewProps = ExerciseGroup;

export class GroupView extends React.Component<GroupViewProps> {
	public render(): ReactElement {
		return (
			<div>
				<h1>{this.props.name}</h1>
				{this.props.exercises.map(({ exerciseId, duration, count, rest }) => {
					return (
						<div>
							<p>{exerciseId}</p>
							<p>Duration: {duration}</p>
							<p>Count: {count}</p>
							<p>Name: {rest}</p>
						</div>
					);
				})}
				<p>Repetitions: {this.props.repetitions}</p>
				<p>Rest: {this.props.rest}</p>
			</div>
		);
	}
}
