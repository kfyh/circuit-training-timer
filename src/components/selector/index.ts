export * from './CircuitListView';
export * from './CircuitView';
