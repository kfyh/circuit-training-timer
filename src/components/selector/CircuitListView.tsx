import React, { ReactElement } from 'react';
import Grid from '@material-ui/core/Grid';
import { Circuit } from '../../types/circuits';
import { CircuitView } from './CircuitView';

type CircuitListViewProps = {
	circuits: Array<Circuit>;
	onSelect: (circuit: Circuit) => void;
	onEdit: (circuit: Circuit) => void;
};

export const CircuitListView = ({ circuits, onSelect, onEdit }: CircuitListViewProps): ReactElement => {
	return (
		<Grid container spacing={2}>
			{circuits.map((circuit) => {
				return (
					<Grid item key={circuit.id}>
						<CircuitView key={circuit.id} circuit={circuit} onSelect={onSelect} onEdit={onEdit} />
					</Grid>
				);
			})}
		</Grid>
	);
};
