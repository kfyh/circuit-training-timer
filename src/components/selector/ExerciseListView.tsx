import React, { ReactElement } from 'react';
import { Exercise } from '../../types/circuits';
import { ExerciseView } from './ExerciseView';

type ExerciseListViewProps = {
	exercises: Array<Exercise>;
	onSelect: (exercise: Exercise) => void;
};

export const ExerciseListView = ({ exercises, onSelect }: ExerciseListViewProps): ReactElement => {
	return (
		<div>
			{exercises.map((exercise) => {
				return <ExerciseView key={exercise.id} exercise={exercise} onClick={onSelect} />;
			})}
		</div>
	);
};
