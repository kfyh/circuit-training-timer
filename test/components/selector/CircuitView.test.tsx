import React from 'react';
import { shallow } from 'enzyme';
import { CircuitView } from '../../../src/components/selector/CircuitView';
import { circuit } from '../../fixtures/exerciseFixtures';

test('Given CircuitView, it renders correctly', () => {
	const render = shallow(<CircuitView circuit={circuit} onClick={jest.fn()} />);
	expect(render).toMatchSnapshot();
});

test('Given CircuitView, when button click, then onClick called', () => {
	const mockCallback = jest.fn();
	const render = shallow(<CircuitView circuit={circuit} onClick={mockCallback} />);

	const event = { preventDefault: () => {} };
	render.find('button').simulate('click', event);

	expect(mockCallback).toHaveBeenCalledWith(circuit);
});
