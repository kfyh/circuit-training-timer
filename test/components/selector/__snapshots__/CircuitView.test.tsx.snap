// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given CircuitView, it renders correctly 1`] = `
<div>
  circuit
  <button
    onClick={[Function]}
  >
    Select
  </button>
</div>
`;
